stream_alert/alert_processor/outputs/github.py

```diff
@@ -0,0 +1,107 @@
+
+from stream_alert.alert_processor import LOGGER
+from stream_alert.alert_processor.outputs.output_base import (
+    OutputDispatcher,
+    OutputProperty,
+    OutputRequestFailure,
+    StreamAlertOutput
+)
+
+
+@StreamAlertOutput
+class GithubOutput(OutputDispatcher):
+    """GithubOutput handles all alert dispatching for Github"""
+    __service__ = 'github'
+
+    @classmethod
+    def get_user_defined_properties(cls):
+        """Get properties that must be assigned by the user when configuring a new Github output."""
+        return OrderedDict([
+            ('descriptor',
+             OutputProperty(description='a short and unique descriptor for this'
+                                        ' Github integration')),
+            ('repository',
+             OutputProperty(description='the repository for this integration '
+                                        'in the form :username/:repository',
+                            cred_requirement=True,
+                            mask_input=False)),
+            ('labels',
+             OutputProperty(description='a comma separated list of labels to '
+                                        'apply to issues when they are created',
+                            cred_requirement=True,
+                            mask_input=False)),
+            ('username',
+             OutputProperty(description='the username for this integration',
+                            cred_requirement=True,
+                            mask_input=False)),
+            ('access_token',
+             OutputProperty(description='the access token for the integration',
+                            cred_requirement=True,
+                            mask_input=True))
+        ])
+
+    @classmethod
+    def _get_default_properties(cls):
+        """Get the standard url used for Github API. This value the same for
+        everyone, so is hard-coded here and does not need to be configured by
+        the user
+
+        Returns:
+            dict: Contains various default items for this output (ie: url)
+        """
+        return {'api': 'https://api.github.com'}
+
+    def dispatch(self, **kwargs):
+        """Send alert to Github
+
+        Args:
+            **kwargs: consists of any combination of the following items:
+                descriptor (str): Service descriptor (ie: slack channel, pd integration)
+                rule_name (str): Name of the triggered rule
+                alert (dict): Alert relevant to the triggered rule
+        """
+        credentials = self._load_creds(kwargs['descriptor'])
+        if not credentials:
+            return self._log_status(False)
+
+        username_password = "{}:{}".format(credentials['username'],
+                                           credentials['access_token'])
+        encoded_credentials = base64.b64encode(username_password)
+        headers = {'Authorization': "Basic {}".format(encoded_credentials)}
+        url = os.path.join(credentials['api'],
```

Comment:
using `os.path.join` here wouldn't be a problem for production since this runs in lambda, but it's worth noting that if someone is testing this on Windows it will produce an invalid url. can you instead just use format for the whole url like:

`url = '{}/repos/{}/issues'.format(credentials['api'], credentials['repository'])`